test/e2e/DeploymentSetUp.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+// solhint-disable reason-string
+// solhint-disable not-rely-on-time
+// solhint-disable avoid-low-level-calls
+
+import "forge-std/Test.sol";
+import "./DeploymentConstants.sol";
+import "./ICoWSwapSettlementExtended.sol";
+import "../Constants.sol";
+import "../../src/interfaces/IWrappedNativeToken.sol";
+
+contract DeploymentSetUp is Test {
+    struct Contracts {
+        ICoWSwapSettlementExtended settlement;
+        IWrappedNativeToken weth;
+    }
+
+    function deploy() public returns (Contracts memory) {
+        ICoWSwapSettlementExtended settlement = deployCoWSwapContracts();
+        IWrappedNativeToken weth = deployWrappedNativeToken();
+        return Contracts(settlement, weth);
+    }
+
+    function deployWrappedNativeToken()
+        internal
+        returns (IWrappedNativeToken wrappedNativeToken)
+    {
+        wrappedNativeToken = IWrappedNativeToken(
+            0xE76e76E76E76E76e76e76e76E76E76E76E76e760
+        );
+        // Note: WETH9 has no constructor, so most features will still be working when just copying over the deployed
+        // bytecode.
+        // Bytecode obtained from:
+        vm.etch(address(wrappedNativeToken), DeployedBytecode.WETH9);
+    }
+
+    function deployCoWSwapContracts()
+        internal
+        returns (ICoWSwapSettlementExtended settlement)
+    {
+        address deployer = deployDeterministicDeployer();
+        settlement = ICoWSwapSettlementExtended(Constants.COWSWAP_SETTLEMENT);
+
+        require(address(settlement).code.length == 0);
+        // We deploy the contract in the exact same way as the real deployment.
+        (bool success, ) = deployer.call(DeterministicDeploymentTx.SETTLEMENT);
+        require(success, "Settlement contract deployment failed");
+
+        // Deterministic deployment address confirms that the deployment was successful
+        require(
+            address(settlement).code.length != 0,
+            "Settlement contract not deployed at expected address"
+        );
+
+        // Deploy trivial authenticator that accepts any address as solver.
+        // No storage needs to be set, so we just copy the deployed bytecode.
+        vm.etch(
+            settlement.authenticator(),
+            type(TrivialAuthenticator).runtimeCode
```

Comment:
Hehe - I started doing the same thing in my [`cow-trade-simuation`](https://github.com/nlordell/cow-trade-simulation/blob/main/contracts/AnyoneAuthenticator.sol) repo for trade simulations with state overrides.